src/decoder/mod.rs

```diff
@@ -1139,30 +1101,6 @@ impl<R: Read + Seek> Decoder<R> {
         }
     }
 
-    /// Read a single strip from the image and return it as a Vector. This method does not return
-    /// information on the size of the strip read and so the caller must take care of the final
-    /// strip in the image as the data read may be less than the expected strip size. This interface
-    /// may change at a future date to reflect this.
-    #[deprecated = "Use read_chunk instead"]
-    pub fn read_strip(&mut self) -> TiffResult<DecodingResult> {
```

Comment:
A bit simpler, here it's only the `chunk_index` to maintain.

Reply:
I'd argue that it is _better_ to have the `chunk_index` get passed in by the user. It is useless for the user to get a chunk of image data without knowing which one it is, so they have to be tracking that information anyway. The new approach avoids the risk of the user's value getting out of sync with the library internal version

Reply:
A good point. My view was from the side of processing a complete image but even that keeps the index implicitly in an iterator. Very well then, let's just see if someone comes up with a more useful and appropriate abstraction in the new interface.